src/Orleans.Runtime/MembershipService/ClusterHealthMonitor.cs

```diff
@@ -346,8 +342,9 @@ Task OnActiveStop(CancellationToken ct)
 
                 this.monitoredSilos = this.monitoredSilos.Clear();
 
-                // Stop waiting for graceful shutdown when the provided cancellation token is cancelled
-                return Task.WhenAny(ct.WhenCancelled(), Task.WhenAll(tasks));
+                // Allow some minimum time for graceful shutdown.
+                var gracePeriod = Task.WhenAll(Task.Delay(TimeSpan.FromSeconds(5)), ct.WhenCancelled());
```

Comment:
Should we make the grace period configurable? Or at the very minimum make it a constant as it is used multiple times below?

Reply:
Making it configurable may be too much ceremony for too little benefit. I'll make it a constant

Reply:
Fixed